File: UsingDagger2/app/src/main/java/sample/github/nisrulz/usingdagger2/MainFragment.java
package sample.github.nisrulz.usingdagger2;

import android.os.Bundle;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import javax.inject.Inject;

/**
 * A simple {@link Fragment} subclass.
 */
public class MainFragment extends Fragment implements MainView {

  @Inject MainPresenter presenter;

  @Override public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    DaggerMainComponent.builder().mainModule(new MainModule(this)).build().inject(this);
  }

  public MainFragment() {
    // Required empty public constructor
  }

  @Override public View onCreateView(LayoutInflater inflater, ViewGroup container,
      Bundle savedInstanceState) {
    // Inflate the layout for this fragment
    return inflater.inflate(R.layout.fragment_main, container, false);
  }
}

File: UsingDagger2/app/src/main/java/github/nisrulz/example/usingdagger2/MainFragment.kt
package github.nisrulz.example.usingdagger2

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import github.nisrulz.example.usingdagger2.databinding.FragmentMainBinding
import javax.inject.Inject

class MainFragment : Fragment(), MainView {

    private lateinit var binding: FragmentMainBinding

    private val module by lazy { MainModule(this) }

    var presenter: MainPresenter? = null
        @Inject set

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Setup Dagger
        DaggerMainComponent
            .builder()
            .mainModule(module)
            .build()
            .inject(this)
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        // Use the inflater to inflate the layout via the Binding class
        binding = FragmentMainBinding.inflate(inflater, container, false)
        return binding.root
    }
}